src/sources/aws_s3/sqs.rs

```diff
@@ -530,77 +541,34 @@ impl IngestorProcess {
             None => lines,
         };
 
-        let mut stream = lines.map(|line| {
-            let deserializer = BytesDeserializer::new();
-            let mut log = deserializer
-                .parse_single(line, log_namespace)
-                .with_batch_notifier_option(&batch);
-
-            log_namespace.insert_source_metadata(
-                AwsS3Config::NAME,
-                &mut log,
-                Some(LegacyKey::Overwrite(path!("bucket"))),
-                path!("bucket"),
-                Bytes::from(s3_event.s3.bucket.name.as_bytes().to_vec()),
-            );
-            log_namespace.insert_source_metadata(
-                AwsS3Config::NAME,
-                &mut log,
-                Some(LegacyKey::Overwrite(path!("object"))),
-                path!("object"),
-                Bytes::from(s3_event.s3.object.key.as_bytes().to_vec()),
-            );
-            log_namespace.insert_source_metadata(
-                AwsS3Config::NAME,
-                &mut log,
-                Some(LegacyKey::Overwrite(path!("region"))),
-                path!("region"),
-                Bytes::from(s3_event.aws_region.as_bytes().to_vec()),
-            );
-
-            if let Some(metadata) = &metadata {
-                for (key, value) in metadata {
-                    log_namespace.insert_source_metadata(
-                        AwsS3Config::NAME,
-                        &mut log,
-                        Some(LegacyKey::Overwrite(key.as_str())),
-                        path!("metadata", key.as_str()),
-                        value.clone(),
-                    );
+        let mut stream = lines.flat_map(|line| {
+            let events = match self.state.decoder.deserializer_parse(line) {
+                Ok((events, _events_size)) => events,
+                Err(error) => {
+                    // Error is logged by `codecs::Decoder`, no further handling
```

Comment:
>  Error is logged by `codecs::Decoder`, no further handling

nit- is an internal error event emitted? In which case you might change the word "logged" to "handled" to be more clear